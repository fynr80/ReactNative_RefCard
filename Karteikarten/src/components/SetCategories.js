/**
 * @file Get and Return Categories & Style for HomeScreen
 * @author Fatih Yanar fatih_yanar@gmx.de
 */

import * as React from 'react';
import {
    View,
    Text,
    TouchableOpacity,
    ImageBackground
} from "react-native";
import {useEffect, useState} from "react";

//import {LinearGradient} from 'expo-linear-gradient';
import firebase from "firebase/compat";
import {useNavigation} from '@react-navigation/native';
import IconF from 'react-native-vector-icons/FontAwesome'

import stylejs from "../style";


const SetCategories = (props) => {
    const id = props.id
    const categoriesName = props.name
    const categoryColor = props.categoryColor;
    const navigation = useNavigation();
    const [themeCounter, setThemeCounter] = useState(0);


    useEffect(() => {
        return getThemeCounter()
    })

    /**
     * Get the amount of Themes of each Category
     */
    const getThemeCounter = () => {
        firebase.firestore()
            .collection('category')
            .doc(props.id)
            .collection("set")
            .onSnapshot(snapshot => {
                const arr = snapshot.docs.map((doc) => ({
                    id: doc.id,
                    name: doc.data().titel
                }))
                setThemeCounter(arr.length)
            })
    }

    /**
     * Got the SearchScreen of the selected Category
     */
    const goToSearch = () => {
        navigation.navigate("Search", {id, categoriesName, categoryColor})
    }

    return (
        <TouchableOpacity
            onPress={goToSearch}
        >
            <View testID="homescreenSet" style={[stylejs.items_Container,{height:120, backgroundColor: categoryColor.left}]}>
            {/* <LinearGradient style={[stylejs.items_Container,{height:120}]} colors={[categoryColor.left,categoryColor.right]}>*/}
                <ImageBackground  source={require('../asserts/intelligence.png')} imageStyle={{width:'30%',position:'absolute',left:100,opacity:0.3}} style={{width: '100%', height: '100%',flexDirection:'row',justifyContent:'space-between'}}>
                    <View >
                        <Text testID={"homescreenSetTitle"} numberOfLines={2} style={stylejs.items_titleText}>{props.name}</Text>
                        <Text style={stylejs.items_amountText}>{themeCounter} Sets</Text>
                    </View>
                    <View>
                        <IconF name='chevron-right' size={35} color='white'/>
                    </View>
                </ImageBackground>
            {/*  </LinearGradient>*/}
            </View>
        </TouchableOpacity>
    )
};

export default SetCategories;
