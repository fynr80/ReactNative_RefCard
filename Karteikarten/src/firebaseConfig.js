
import firebase from 'firebase/compat/app';

const firebaseConfig = {
    apiKey: "AIzaSyBksswFLr1YB6en-sRf8mqSFle7bhAzSgU",
    authDomain: "swtp-812df.firebaseapp.com",
    databaseURL: "https://swtp-812df-default-rtdb.firebaseio.com",
    projectId: "swtp-812df",
    storageBucket: "swtp-812df.appspot.com",
    messagingSenderId: "44236558384",
    appId: "1:44236558384:web:2f15f6b5fb190634b71df5",
    measurementId: "G-W33L0G7D18"
};
firebase.initializeApp(firebaseConfig);
export {firebase};
