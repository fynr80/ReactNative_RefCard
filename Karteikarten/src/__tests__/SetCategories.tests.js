import React from 'react';
import renderer from 'react-test-renderer';
import {fireEvent, render, screen} from '@testing-library/react-native'
import SetCategories from "../components/SetCategories";
import firebaseConfig from "../firebaseConfig";

const mockedNavigation = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
        ...actualNav,
        useNavigation: () => ({
            navigate: mockedNavigation,
            dispatch: jest.fn(),
        }),
    };
});


describe("Category List elements Component", () => {
    it('renders List element correctly',  () => {
        const tree = renderer.create(<SetCategories name='Calculus' id="2" categoryColor=""/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Set title is correctly',  () => {
        const {getByTestId} = render(<SetCategories name='Calculus' id="2" categoryColor=""/>);
        const title = getByTestId("homescreenSetTitle");
        expect(title.props.children).toMatch("Calculus");
    });

    it('Set Color is correctly',  () => {
        const bgcolor = {
            left: "#2B32B2",
            right: '#1488CC',
        }
        const {getByTestId} = render(<SetCategories name='Calculus' id="2" categoryColor={bgcolor}/>);
        const set = getByTestId("homescreenSet");
        expect(set.props.style[1].backgroundColor).toEqual(bgcolor.left);
    });


    it('Calls Navigation with Correct Params',  () => {
        const {getByText} = render(<SetCategories name='Calculus' id="2" categoryColor=""/>);
        const button = screen.getByText("Calculus");
        fireEvent.press(button);
        expect(mockedNavigation).toBeCalledWith("Search", {
            "categoriesName": "Calculus",
            "categoryColor": "",
            "id": "2"
        });
    });
})
