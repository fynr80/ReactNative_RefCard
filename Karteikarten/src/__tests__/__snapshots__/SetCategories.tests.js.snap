// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category List elements Component renders List element correctly 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 30,
          "flexDirection": "row",
          "height": 180,
          "justifyContent": "space-between",
          "marginTop": 20,
          "paddingLeft": 15,
          "paddingRight": 15,
          "paddingTop": 20,
          "width": "100%",
        },
        Object {
          "backgroundColor": undefined,
          "height": 120,
        },
      ]
    }
    testID="homescreenSet"
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "flexDirection": "row",
          "height": "100%",
          "justifyContent": "space-between",
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": "100%",
              "width": "100%",
            },
            Object {
              "left": 100,
              "opacity": 0.3,
              "position": "absolute",
              "width": "30%",
            },
          ]
        }
      />
      <View>
        <Text
          numberOfLines={2}
          style={
            Object {
              "color": "white",
              "fontSize": 20,
              "fontWeight": "bold",
              "maxWidth": 300,
              "paddingRight": 30,
            }
          }
          testID="homescreenSetTitle"
        >
          Calculus
        </Text>
        <Text
          style={
            Object {
              "color": "white",
              "fontSize": 15,
              "opacity": 0.8,
            }
          }
        >
          0
           Sets
        </Text>
      </View>
      <View>
        <Text />
      </View>
    </View>
  </View>
</View>
`;
